torch/testing/_internal/opinfo/core.py

```diff
@@ -943,29 +954,61 @@ def __post_init__(self):
 
         self.dtypes = set(self.dtypes)
 
+        self.dtypesIfGPU = (
+            set(self.dtypesIfGPU) if self.dtypesIfGPU is not None else self.dtypes
+        )
+        self.dtypesIfCUDA = (
```

Comment:
We have defined the `dtypesIfGPU`, why is the `dtypesIfCUDA` still required? IIUC, the design leaves flexibility to users in case the `dtypesIfGPU` cannot cover all test cases. @daisyden , right?

Reply:
Your understanding is right. The dtypesIfCUDA or dtypesIfXPU would be used if a device supports some specific dtypes that is not covered by dtypesIfGPU.

Reply:
Let me add some comments here.